releasenotes/notes/fix-staged-passmanager-c2e423ebf17eb57e.yaml

```diff
@@ -0,0 +1,5 @@
+---
+other:
+  - |
+    Updates documentaiton on how repeated stage names are handled in
+    :class:`qiskit.transpiler.StagedPassManager`, and adds test to check for correctness.
```

Comment:
We don't really need a release note for this now. The intent of release notes is to document any changes in the release for users. Updated documentation and tests don't really effect usage of qiskit and since there is no behavior change anymore there really isn't anything we need to document for users when they're upgrading to a newer version with this PR included.